src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,566 @@
+        }
+
+        DiscordInteractionInstruction instructionEnum = DiscordInteractionInstruction.valueOf(instruction.asString().toUpperCase());
+        if (!DenizenDiscordBot.instance.connections.containsKey(id.asString())) {
+            Debug.echoError("Failed to process DiscordMessage command: unknown bot ID!");
+            scriptEntry.setFinished(true);
+            return;
+        }
+        JDA client = DenizenDiscordBot.instance.connections.get(id.asString()).client;
+        boolean isEphermal = ephermal == null ? false : ephermal.asBoolean();
+
+        Bukkit.getScheduler().runTaskAsynchronously(DenizenDiscordBot.instance, () -> {
+            switch (instructionEnum) {
+                case COMMAND: {
+                    if (commandInstruction == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must have a command instruction!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    } 
+                    else if (name == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must specify a name!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+
+                    DiscordInteractionCommandInstruction commandInstructionEnum = DiscordInteractionCommandInstruction.valueOf(commandInstruction.asString().toUpperCase());
+                    boolean isEnabled = enabled == null ? true : enabled.asBoolean();
+
+                    switch (commandInstructionEnum) {
+                        case CREATE: {
+                            if (description == null) {
+                                Debug.echoError(scriptEntry.getContext(), "Must specify a description!");
+                                scriptEntry.setFinished(true);
+                                return;
+                            }
+                            CommandData data = new CommandData(name.asString(), description.asString());
+                            if (options != null) {
+                                for (MapTag option : options) {
+                                    ElementTag typeStr = (ElementTag) option.getObject("type");
+                                    if (typeStr == null) {
+                                        Debug.echoError(scriptEntry.getContext(), "Command options must specify a type!");
+                                        scriptEntry.setFinished(true);
+                                        return;
+                                    }
+                                    OptionType optionType = null;
+                                    for (OptionType val : OptionType.values()) {
+                                        if (val.name().toUpperCase().replace("_", "").equals(
+                                            typeStr.asString().toUpperCase().replace("_", "")
+                                        )) {
```

Comment:
daheck happened here with formatting

Reply:
honestly not sure

Reply:
also why is this not a `valueOf`

Reply:
I observed other enum matches and I think I found something similar to this, iterating over every one until it finds a good match. Would there be any problems with valueOf? And would I have to catch exceptions for it?

Reply:
I don't think Denizen does enum matching like that anywhere, with the exception of pseudo-enums and some not-lookup matching. Refer to... any enum gets, including the other ones in this same class file lol, for how to do valueOf

Reply:
The only enum gets I see are checked with Argument#matchesEnum calls (in parseArgs). Is there another way to do this? The ArgumentHelper class doesn't seem to have a matchesEnum method.